### PR TITLE
Add support for widget image

### PR DESCRIPTION
**Pull request by Puncher1**
## Summary

This PR adds a new method `widget_image_url` to `Guild`, which returns an URL to the widget image.

## Checklist

- [x] If code changes were made then they have been tested.
    - [x] I have updated the documentation to reflect the changes.
- [ ] This PR fixes an issue.
- [x] This PR adds something new (e.g. new method or parameters).
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed)
- [ ] This PR is **not** a code change (e.g. documentation, README, ...)

### CODE REVIEW

File: discord/guild.py [hunk 1]
```diff
@@ -3931,6 +3932,38 @@ async def widget(self) -> Widget:
 
         return Widget(state=self._state, data=data)
 
+    WidgetStyle = Literal['shield', 'banner1', 'banner2', 'banner3', 'banner4']
```
Comment (Gobot1234) [inline discord/guild.py:3935]: I wouldn't advise putting this in the namespace as then you can do Guild.WidgetStyle which sits a little weird with me

File: discord/guild.py [hunk 2]
```diff
@@ -3931,6 +3933,36 @@ async def widget(self) -> Widget:
...
+        Parameters
+        -----------
+        style: :class:`str`
+            The style which should be applied for the widget.
+            Default to ``shield``. Returns ``None`` if widget is not enabled.
+
+        Raises
+        -------
+        HTTPException
+            Retrieving the widget image url failed.
```
Comment (Rapptz) [inline discord/guild.py:3954]: This doesn't raise anything.

**Review by Rapptz — COMMENTED** [review]
I'm still unsure what the proper name for this is, since it's inconsistent with other methods that are similar, i.e. `Asset` but we talked about this in the discord server.

File: discord/guild.py (hunk 2 above)
Comment (Puncher1) [reply discord/guild.py]: Ah okay. Well it does when the user specifies a wrong `style` but that's not in the bot's scope anymore, the link just returns an error in the browser.

**Comment by Puncher1** [comment]
Yes. I didn't found a good name for it. I first thought about `widget_image_link` but idk if this is the way to go.

**Comment by Puncher1** [comment]
How about turning this into an helper function in `utils.py` like `oauth_url`?

**Comment by Puncher1** [comment]
Closing this because it doesn't seem to be useful. Can be reopened again if this changes.

**Pull request closed without merging**